framework/test/unit/engine/consensus/certificate_generation/commit_pool.spec.ts

```diff
@@ -1380,5 +1410,19 @@ describe('CommitPool', () => {
 				certificateSignature: Buffer.alloc(0),
 			});
 		});
+
+		it('should certify the minCertifyHeight even when heightNextBFTParameters is lower', async () => {
+			// Arrange
+			bftMethod.getNextHeightBFTParameters.mockResolvedValue(309);
+			(commitPool['_minCertifyHeight'] as any) = heightNextBFTParameters - 1;
```

Comment:
Just to confirm, here we set `_minCertifyHeight` to 1052 and then expect the selected aggregate commit to have height 1052 and not 1051? This is correct, we could maybe additionally also test that if `_minCertifyHeight == 1051` then we get `singleCommit1`, i.e., the selected aggregate commit has height 1051.

Reply:
Isn't the test in line 1428 and line 1442 now the same? I could not spot the difference.

Reply:
Looks good now.